src/vivarium/framework/event.py

```diff
@@ -111,7 +90,7 @@ def __init__(self, manager):
         self.manager = manager
         self.listeners = [[] for _ in range(10)]
 
-    def emit(self, event: Event) -> Event:
+    def emit(self, index: pd.Index, user_data: Dict = None) -> Event:
```

Comment:
I think defaulting user_data to None and never setting it to an empty dict if it is None in the code itself (as was previously done in line 58 in the old code) may break things that e.g., check ``if 'sim_state' in event.user_data``

Reply:
actually I'm not positive if it will break things. It may be that anything that does a check like that will check first if user_data is None - I'm not sure. but since previously we had the user_data always being an empty dict, it might be best just to keep that convention

Reply:
Definitely go ahead and coerce Nones to empty dicts

Reply:
having it non-mutable is better practice, though. Maybe we should work out the kinks